src/node-lib/curriculum-api-2023/queries/specialistUnitListing/specialistUnitListing.query.ts

```diff
@@ -0,0 +1,208 @@
+          subjectTitle: unit.combined_programme_fields.subject,
+          expired: unit.expired || false,
+          expiredLessonCount:
+            batchRes.specialistUnitExpiredLessonCount.aggregate.count,
+          themeSlug: unit.combined_programme_fields.phase_slug || null,
+          themeTitle: unit.combined_programme_fields.phase || null,
+          learningThemes: [
+            {
+              themeSlug: unit.combined_programme_fields.phase_slug,
+              themeTitle: unit.combined_programme_fields.phase,
+            },
+          ],
+          developmentalStageSlug:
+            unit.combined_programme_fields.developmentstage_slug,
+          developmentalStageTitle:
+            unit.combined_programme_fields.developmentstage,
+        },
+      ];
+    }
+  });
+
+  return units;
+};
+
+export const getExpandedDevelopmentalStages = (
+  partialDevelopmentStages: Array<Partial<DevelopmentalStage>>,
+  data: BatchResultResponseArray,
+) => {
+  return partialDevelopmentStages.map((stage, i) => {
+    const batchRes = developmentStageUnitCount.parse(data[i]?.data);
+    if (data[i]?.data) {
+      return {
+        ...stage,
+        unitCount: batchRes.developmentStageUnitCount.aggregate.count,
+        lessonCount: batchRes.developmentStageLessonCount.aggregate.count,
+      };
+    }
+  });
+};
+
+export const getUnitBatchRequests = (
+  specialistUnits: SpecialistUnitListRequestSchema,
+) => {
+  return specialistUnits.map((c) => {
+    return {
+      document: SpecialistLessonCountDocument,
+      variables: { unit_slug: c.unit_slug },
+    };
+  });
+};
+
+export const getDevelopmentStagesBatchRequests = (
+  developmentalStages: Partial<DevelopmentalStage>[],
+) => {
+  return developmentalStages.map((c) => {
+    return {
+      document: DevelopmentStageUnitCountDocument,
+      variables: {
+        _contains: {
+          developmentstage_slug: c?.slug,
+        },
+        syntheticProgrammeSlug: c?.programmeSlug,
+      },
+    };
+  });
+};
+
+export const getPartialDevelopmentStages = (
+  specialistUnits: SpecialistUnitListRequestSchema,
+) => {
+  return specialistUnits.reduce(
+    (acc, stage) => {
+      const stageSlug = stage.combined_programme_fields.developmentstage_slug;
+      const stageTitle = stage.combined_programme_fields.developmentstage;
+      if (stageSlug && stageTitle && !acc.find((s) => s?.slug === stageSlug)) {
+        const developmentStage = {
+          slug: stageSlug,
+          title: stageTitle,
+          programmeSlug: stage.synthetic_programme_slug,
+        };
+        acc.push(developmentStage);
+      }
+      return acc;
+    },
+    [] as Array<Partial<DevelopmentalStage>>,
+  );
+};
+
+export const getThemes = (specialistUnits: SpecialistUnitListRequestSchema) => {
+  return specialistUnits.reduce(
+    (acc, unit) => {
+      const themeSlug = unit.combined_programme_fields.phase_slug;
+      const themeTitle = unit.combined_programme_fields.phase;
```

Comment:
oh this is interesting, so there is a column `threads` in the specialist MV that I would have thought would be the `themes` to show on the unit page: 
<img width="478" alt="Screenshot 2024-03-13 at 09 28 27" src="https://github.com/oaknational/Oak-Web-Application/assets/45038878/9c1d542d-b39c-4e57-b51b-44c74f205fc7">

Reply:
Ah was going on the unit.view that used the phase as theme.